app/models/payment.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+# Payments made by an agency to a provide for a particular site
+class Payment < UuidApplicationRecord
+  belongs_to :agency
+  belongs_to :site
+
+  validates :amount, numericality: { greater_than: 0.00 }
+  validates :care_finished_on, presence: true # TODO: use Date validator
+  validates :care_started_on, presence: true # TODO: use Date validator
+  validates :discrepancy, numericality: true
+  validates :paid_on, presence: true # TODO: use Date validator
+
+  before_validation { |payment| payment.slug = generate_slug("#{payment.care_started_on}#{payment.site_id}#{payment.agency_id}") }
+
+  # The money-rails gem specifically requires that the '_cents' suffix be
+  # specified when using the "monetize" macro even though the attributes are
+  # referred to without the '_cents' suffix.
+  # IOW, you only need to refer to payment.amount or payment.discrepancy ,
+  # unlike the following statements.
```

Comment:
![https://media.giphy.com/media/l2JJOD8cdCuFEPHxe/giphy.gif](https://media.giphy.com/media/l2JJOD8cdCuFEPHxe/giphy.gif)

Reply:
That's part of working with the `money` gem.  I thought it was worth noting so that future readers would be clued in to that.  It's definitely something that they should fix with the gem so they're consistent everywhere.  This is often a source of confusion.

Reply:
My gif reaction was to the money gem itself, not to you or your comments (which are wonderful!) 😂